megamek/src/megamek/common/Board.java

```diff
@@ -1894,4 +1897,90 @@ public Set<String> getTags() {
     public String getMapName() {
         return "Board #" + boardId;
     }
+    
+    /**
+     * @return Given an "exits" value, returns it in a list form.
+     * (i.e. exits value of 4 returns {3}, exit value of 5 returns {1, 3}
+     */
+    public static List<Integer> exitsAsIntList(int exits) {
+        List<Integer> results = new ArrayList<>();
+        
+        for (int bitToCheck = 1, exitIndex = 1; bitToCheck <= 16536; bitToCheck *= 2, exitIndex++) {
+            if ((exits & bitToCheck) != 0) {
+                results.add(exitIndex);
+            }
+        }
+        
+        return results;
+    }
+    
+    /**
+     * Worker function that initializes any custom deployment zones present on the board
+     */
+    private void initializeDeploymentZones() {
+        deploymentZones = new HashMap<>();
+        
+        for (int x = 0; x < getWidth(); x++) {
+            for (int y = 0; y < getHeight(); y++) {
+                Hex currentHex = getHex(x, y);
+                Terrain deploymentZone = currentHex.getTerrain(Terrains.DEPLOYMENT_ZONE);
+                
+                if (deploymentZone != null) {
+                    for (int zoneID : Board.exitsAsIntList(deploymentZone.getExits())) {
+                        if (!deploymentZones.containsKey(zoneID)) {
```

Comment:
deploymentZones.computeIfAbsent(zoneID, k->new HashSet<>()).add(new Coords(x,y);
is 1930-1934
:)